rust/agama-cli/src/profile.rs

```diff
@@ -9,32 +10,59 @@ use std::{
     process::Command,
 };
 use tempfile::TempDir;
+use url::Url;
 
 #[derive(Subcommand, Debug)]
 pub enum ProfileCommands {
-    /// Download the profile from a given location
-    Download { url: String },
+    /// Download the autoyast profile and print resulting json
+    Autoyast {
+        /// URL to autoyast XML, erb or rules&classes dir. Supports
+        /// all schemas that autoyast supports.
+        url: String,
+    },
 
     /// Validate a profile using JSON Schema
-    Validate { path: String },
+    ///
+    /// Schema is available at /usr/share/agama-cli/profile.schema.json
+    Validate {
+        /// Local path to json file
+        path: String,
+    },
 
     /// Evaluate a profile, injecting the hardware information from D-Bus
-    Evaluate { path: String },
+    ///
+    /// For example of jsonnet profile see
+    /// https://github.com/openSUSE/agama/blob/master/rust/agama-lib/share/examples/profile.jsonnet
+    Evaluate {
+        /// Path to jsonnet file.
+        path: String,
+    },
 
     /// Process autoinstallation profile and loads it into agama
     ///
     /// This is top level command that do all autoinstallation processing beside starting
     /// installation. Unless there is a need to inject additional commands between processing
     /// use this command instead of set of underlying commands.
-    /// Optional dir argument is location where profile is processed. Useful for debugging
-    /// if something goes wrong.
-    Import { url: String, dir: Option<PathBuf> },
+    Import {
+        /// URL where profile is located. Supported schemas are all that download supports and additionally
+        /// autoyast specific ones. Supported files are json, jsonnet, sh for agama profiles and erb, xml, directory
+        /// for autoyast support.
+        url: String,
+        /// Specific directory where all processing happens. If not specific temporary directory is used
+        dir: Option<PathBuf>,
+    },
 }
 
-fn download(url: &str, mut out_fd: impl Write) -> anyhow::Result<()> {
-    let reader = ProfileReader::new(url)?;
-    let contents = reader.read()?;
-    out_fd.write_all(contents.as_bytes())?;
+pub fn download(url: &str, mut out_fd: impl Write) -> anyhow::Result<()> {
+    let mut handle = Easy::new();
+    handle.url(url)?;
+
+    let mut transfer = handle.transfer();
+    transfer.write_function(|buf|
+        // unwrap here is ok, as we want to kill download if we failed to write content
+        Ok(out_fd.write(buf).unwrap()))?;
```

Comment:
Do we get a proper error? Not tried, just asking...

Reply:
in debug build for sure...regarding release I have to try it